numba/npyufunc/parfor.py

```diff
@@ -358,12 +370,26 @@ def call_parallel_gufunc(lowerer, cres, gu_signature, outer_sig, expr_args, loop
     if config.DEBUG_ARRAY_OPT:
         print("parallel function = ", wrapper_name, cres)
 
-    if config.DEBUG_ARRAY_OPT:
-        cgutils.printf(builder, "loop_ranges = ")
-        for v in loop_ranges:
-            print("call_parallel_gufunc v = ", v, " ", lowerer.loadvar(v))
-            cgutils.printf(builder, "%d ", lowerer.loadvar(v))
-        cgutils.printf(builder, "\n")
+    # loadvars for loop_ranges
+    def load_range(v):
+        if isinstance(v, ir.Var):
+            return lowerer.loadvar(v.name)
+        else:
+            return context.get_constant(types.intp, v)
+
+    num_dim = len(loop_ranges)
+    for i in range(num_dim):
+        start, stop, step = loop_ranges[i]
+        start = load_range(start)
+        stop = load_range(stop)
+        assert(step == 1) # We do not support loop steps other than 1
```

Comment:
Slicing doesn't need an exception since it is just ignored while other parts of the function are parallelized.

Checking broadcast semantics is a bigger issue in ParallelAccelerator. Let's open a ticket for it and start thinking about solutions.